### ISSUE TITLE
Global users on webknossos.org

### ISSUE THREAD

**Issue by normanrz** [scalableminds/webknossos#4761]
I think the current organization and user account structure is still confusing for new users. I mean that user accounts are tied to organizations, but you can't have the same email address in multiple organizations and you cannot easily share data/annotations across organizations.

I think a global user model such as Github has, would be clearer: People sign up for a user account on webknossos.org, can create new organizations, and invite others to join their organization. This would be a model that most users are familier with. Perhaps we could utilize the team capabilities of wk for that?

@fm3 @philippotto @hotzenklotz  thoughts?

**Comment by philippotto** [comment]
Conceptionally, this approach sounds very reasonable to me. Implementation-wise I'm not sure how we should solve this (and how much effort it would be). From my understanding, organizations are very strictly separated right now and we would need to weaken this (without introducing privacy problems). Probably needs a lot of care..

Also, I'm not sure whether it's worth to implement it right now. Do you think, there are many users who would (like to) share stuff across organizations or be member of multiple organizations?

Should we discuss this in person?

**Comment by normanrz** [comment]
I think it would be good to understand what it would take to implement this.

A problem that I see frequently, is that users create new orgs but probably want to join an existing organization instead. Maybe they don't know that at the point of signing up yet, though.

**Comment by fm3** [comment]
A cheaper solution for the described workflow could be a functionality of “merging” organizations. If a user signs up but notices they actually wanted to join another organization, they are usually the only person in their orga. We could allow an operation that moves their existing datasets / annotations / teams / user into another organization when they join it, and delete the old (single-user) organization.

That way we wouldn’t have to adapt every query and make the user chose for which of their organizations each operation should be performed.

A downside would be that team names and dataset names need to be unique at the time of that merging operation.

**Comment by philippotto** [comment]
@normanrz @fm3 @youri-k and I discussed this topic today. The current plan is to add another indirection from the "authenticated user" to specific "organization-"user which is picked by remembering the "active organization" (e.g., via a cookie). That way, we don't have to adapt every DB query as it's always clear which organization should be used. Also, the UI would only need minimal adaption.

The different steps for this issue would be:

- [ ] Be able to be member of different organizations (but one is always marked as active)
- [ ] Allow being invited to another organization (e.g., via link)
- [ ] Allow to switch between different organizations (shouldn't need re-authentication)

Further follow-up issues are (I'll add the links in a bit):
- Share user settings across different organization accounts
- Allow to merge organizations (in case one was created accidentally)
- Allow users to see organizations if their email domain matches (opt-in)

Let me know in case I got something wrong or forgot something.
